state-chain/pallets/cf-elections/src/electoral_systems/block_witnesser/block_processor.rs

```diff
@@ -54,18 +52,16 @@ use sp_std::{collections::btree_map::BTreeMap, fmt::Debug, marker::PhantomData,
 	T::Event: Encode,
 	T::Rules: Encode,
 	T::Execute: Encode,
-	T::SafetyMargin: Encode,
 ))]
 pub struct BlockProcessor<T: BWProcessorTypes> {
-	/// A mapping from block numbers to their corresponding block data and the next age to be
-	/// processed. The "age" represents the block height difference between head of the chain and
-	/// block that we are processing, and it's used to know what rules have already been processed
-	/// for such block
-	pub blocks_data: BTreeMap<T::ChainBlockNumber, (T::BlockData, u32)>,
-	pub reorg_events: BTreeMap<T::ChainBlockNumber, Vec<T::Event>>,
+	/// A mapping from block numbers to their corresponding block data, the next age to be
+	/// processed and the safety margin. The "age" represents the block height difference between
+	/// head of the chain and block that we are processing, and it's used to know what rules have
+	/// already been processed for such block
+	pub blocks_data: BTreeMap<T::ChainBlockNumber, (T::BlockData, u32, u32)>,
```

Comment:
I think we should introduce a struct to hold the `(T::BlockData, u32, u32)` data. The tuple itself isn't really self-explanatory, unfortunately.